src/rcms/fm/app/level1/HCALxmlHandler.java

```diff
@@ -223,6 +223,10 @@ public String stripExecXML(String execXMLstring, ParameterSet<FunctionManagerPar
       for (StringT maskedApp: maskedAppArray) {
         //logger.info("[JohnLogVector] " + functionManager.FMname + ": about to start masking " + maskedApp.getString());
         String[] maskedAppParts = maskedApp.getString().split("_");
+        if (maskedAppParts.length != 2 || HCALlevelOneEventHandler.stringToInt(maskedAppParts[0]) || !(HCALlevelOneEventHandler.stringToInt(maskedAppParts[1]))){
```

Comment:
I don't think that would work, since the executive xml needs some stripping of "connections" that connect applications in one FM to another. In other words, the level2s need to strip remnants of xml that correspond to connections to applications in other level2s, so it needs to consider stripping of resources that are not only its child resources (AVAILABLE_RESOURCES in the level2s is a list of just that level2s children execs and apps).